src/zarr/testing/store.py

```diff
@@ -42,6 +43,19 @@ def test_store_type(self, store: S) -> None:
         assert isinstance(store, Store)
         assert isinstance(store, self.store_cls)
 
+    def test_store_eq(self, store: S, store_kwargs: dict[str, Any]) -> None:
+        # check self equality
+        assert store == store
+
+        # check store equality with same inputs
+        # asserting this is important for being able to compare (de)serialized stores
+        store2 = self.store_cls(**store_kwargs)
```

Comment:
we should also add a test showing that if `**store_kwargs` is different, then the two stores are not equal. But I don't think this test can be defined in the base class, since it depends on the structure of `store_kwargs`. So no action needed here.